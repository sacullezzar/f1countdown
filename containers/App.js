import React, { Component } from 'react'
import Timer from '../components/timer'
import RaceData from '../components/raceData'

class App extends Component {
    constructor(props) {
        super(props)
        this.state = {
            isLoading: true,
            number: 0,
            raceData: null
        }

        this.handleTimeChange = this.handleTimeChange.bind(this)
    }

    componentDidMount() {
        fetch('https://ergast.com/api/f1/current.json')
            .then(response => response.json())
            .then(response => this.setState({ isLoading: false, raceData: response }))
            .catch(() => {
                console.error(error)
            })
    }

    handleTimeChange(time) {
        const t1 = new Date
        const t2 = new Date(time)
        const date = Math.round((t2 - t1)/1000)
        this.setState({ number: date})
    }

    render () {
        const { number, raceData } = this.state
        const spinnerStyle = {
            position: 'fixed',
            left: '50%',
            top: '50%'
        }
        return (
            <React.Fragment>
                <h1 style={{textAlign: "center"}}>F1 Countdown</h1>
                <Timer time={number}/>
                {!this.state.isLoading && <RaceData raceData={raceData} handleTimeChange={this.handleTimeChange}/>}
            </React.Fragment>
        )
    }
}

export default App