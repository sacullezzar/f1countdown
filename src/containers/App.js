import React, { Component } from 'react'
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Link
} from 'react-router-dom'
import Countdown from '../containers/Countdown'
import RaceSelect from '../components/raceSelect'
import DriversChamp from './DriversChamp'
import TeamChamp from './TeamChamp'

class App extends Component {
    constructor() {
        super()
        this.state = {
            isLoading: true,
            seasonData: null
        }
        this.fetchData = this.fetchData.bind(this)
    }

    componentDidMount() {
        this.fetchData()
    }

    async fetchData() {
        const season = await fetch('https://ergast.com/api/f1/current.json')
        const seasonJson = await season.json()
        this.setState({ isLoading: true, seasonData: seasonJson.MRData.RaceTable.Races })
        const standings = await fetch('https://ergast.com/api/f1/current/driverStandings.json')
        const standingsJson = await standings.json()
        this.setState({ isLoading: true, standingsData: standingsJson.MRData.StandingsTable.StandingsLists })
        const teamStandings = await fetch('https://ergast.com/api/f1/current/constructorStandings.json')
        const teamStandingsJson = await teamStandings.json()
        this.setState({ isLoading: false, teamData: teamStandingsJson.MRData.StandingsTable.StandingsLists})
    }

    render() {
        const { seasonData, standingsData, teamData, isLoading } = this.state
        return (
            <Router>
            <div>
            <nav>
                <ul>
                <li>
                    <Link to="/">Countdown</Link>
                </li>
                <li>
                    <Link to="/races">Previous Races</Link>
                </li>
                <li>
                    <Link to="/driversStandings">Driver's Championship Standings</Link>
                </li>
                <li>
                    <Link to="/constructorsStandings">Constructor's Championship Standings</Link>
                </li>
                </ul>
            </nav>
    
            {/* A <Switch> looks through its children <Route>s and
                renders the first one that matches the current URL. */}
            <Switch>
                <Route path="/races">
                    {seasonData && seasonData.length ? <RaceSelect  seasonData={seasonData}/> : <h2>Loading...</h2>}
                </Route>
                <Route path="/driversStandings">
                    {standingsData && standingsData.length ? <DriversChamp  standingsData={standingsData}/> : <h2>Loading...</h2>}
                </Route>
                <Route path="/constructorsStandings">
                    {teamData && teamData.length ? <TeamChamp teamData={teamData}/> : <h2>Loading...</h2>}
                </Route>
                <Route path="/">
                    {seasonData && seasonData.length ? <Countdown seasonData={seasonData} isLoading={isLoading}/> : <h2>Loading...</h2>}
                </Route>
            </Switch>
            </div>
        </Router>
        )
    }
}

export default App