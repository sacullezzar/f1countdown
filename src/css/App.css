.container {
    display: flex;
}

.title-box {
    width: 'auto';
}
.title {
    font-size: 4em;
    margin: 10,20,30,40;
    background-color: blueviolet;
}