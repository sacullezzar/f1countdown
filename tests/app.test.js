import React from 'react'
import App from '../containers/App.js'
import sinon from 'sinon'

describe('<App />', function () {
    it('should match snapshot', function () {
        const wrapper = shallow(<App />)
        expect(wrapper).toMatchSnapshot()
    })

    it('should match snapshot', function () {
        const handleChangeSpy = sinon.spy(App.prototype, 'handleChange')
        const wrapper = shallow(<App handleChange={handleChangeSpy}/>)
        wrapper.simulate('change')
        expect(wrapper).toMatchSnapshot()
    })
})