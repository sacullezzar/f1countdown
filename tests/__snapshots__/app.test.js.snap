// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<Fragment>
  <div
    className="main"
  >
    <h1
      className="title"
    >
      F1 Countdown
    </h1>
    <div
      className="wrapper"
    >
      <Timer
        time={0}
      />
    </div>
  </div>
</Fragment>
`;

exports[`<App /> tests component did mount with mocked data 1`] = `
<Fragment>
  <div
    className="main"
  >
    <h1
      className="title"
    >
      F1 Countdown
    </h1>
    <div
      className="wrapper"
    >
      <Timer
        time={0}
      />
    </div>
  </div>
</Fragment>
`;

exports[`<App /> tests method handleTimeChange 1`] = `
<Fragment>
  <div
    className="main"
  >
    <h1
      className="title"
    >
      F1 Countdown
    </h1>
    <div
      className="wrapper"
    >
      <Timer
        time={0}
      />
    </div>
  </div>
</Fragment>
`;
